lib/json-api-vanilla/parser.rb

```diff
@@ -82,10 +77,12 @@ def self.build(hash)
             if data.is_a?(Array)
               # One-to-many relationship.
               ref = data.map do |ref_hash|
-                objects[[ref_hash['type'], ref_hash['id']]]
+                klass = prepare_class(ref_hash, superclass, container)
```

Comment:
Maybe we should invoke `prepare_class` only in case when it is needed (when `objects[[ref_hash['type'], ref_hash['id']]]` is falsy)? It seems there is going quite a lot within `prepare_class` and we should avoid this overhead.

Reply:
Yes, that's right.